pkg/build/controller/build/build_controller.go

```diff
@@ -1483,6 +1496,136 @@ func (bc *BuildController) findBuildCAConfigMap(build *buildv1.Build, buildPod *
 	return true, nil
 }
 
+func (bc *BuildController) controllerConfigWorker() {
+	for {
+		if quit := bc.controllerConfigWork(); quit {
+			return
+		}
+	}
+}
+
+func (bc *BuildController) controllerConfigWork() bool {
+	key, quit := bc.controllerConfigQueue.Get()
+	if quit {
+		return true
+	}
+	defer bc.controllerConfigQueue.Done(key)
+
+	err := bc.handleControllerConfig()
+	bc.handleControllerConfigError(err, key)
+	return false
+}
+
+func (bc *BuildController) handleControllerConfig() error {
+	// Make sure that we have synced the controller config store
+	// This will always be false if the controller is not running on an openshift/installer cluster,
+	// and therefore does not have the service cert signing controller.
+	//
+	// TODO: Remove this once all tests run on openshift/installer clusters
+	if !bc.controllerConfigStoreSynced() {
+		glog.V(2).Info("builds.config.openshift.io not synced")
+		glog.V(4).Info("disabling service signing CA injection for builds")
+		bc.injectServiceCA = false
+		return nil
+	}
+	bc.injectServiceCA = true
+	config, err := bc.configLister.Get("cluster")
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if config == nil {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	if len(config.Spec.AdditionalTrustedCA.Name) == 0 {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	additionalCA, err := bc.configMapClient.ConfigMaps("openshift-config").Get(config.Spec.AdditionalTrustedCA.Name, metav1.GetOptions{})
```

Comment:
should be getting this from a configmap listwatcher so we are using a cache.  You already have one defined anyway, right?

Reply:
We have an informer set up for all Kube secrets. It may take a bit more work to add a namespaced informer for `openshift-config`, but I think it's doable.

Reply:
i'd prefer the new cache because otherwise we're going to be caching all the configmaps in the cluster which is a lot.